File: Frontend/trackr-administrador/src/app/shared/http/examen/respuesta.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Respuesta } from '@models/examen/respuesta';

@Injectable({
    providedIn: 'root'
})
export class RespuestaService {
    private dataUrl = 'respuesta/';

    constructor(private http: HttpClient) { }

    public consultarTodosPorReactivo(idReactivo:number): Observable<Respuesta[]> {
        return this.http.get<Respuesta[]>(this.dataUrl + `consultarTodosPorReactivo/${idReactivo}`);
    }

    public consultarParaFormulario(idRespuesta: number): Observable<Respuesta> {
        return this.http.get<Respuesta>(this.dataUrl + `consultar/${idRespuesta}`);
    }

    public agregar(respuesta: Respuesta): Observable<number> {
        return this.http.post<number>(this.dataUrl + 'agregar', respuesta);
    }

    public editar(respuesta: Respuesta): Observable<void> {
        return this.http.put<void>(this.dataUrl + 'editar', respuesta);   
    }

    public eliminar(idRespuesta: number): Observable<void> {
        return this.http.delete<void>(this.dataUrl + `eliminar/${idRespuesta}`);
    }

}

File: Backend/TrackrAPI/TrackrAPI/Services/GestionExamen/RespuestaService.cs

using TrackrAPI.Dtos.GestionExamen;
using TrackrAPI.Models;
using TrackrAPI.Repositorys.GestionExamen;

namespace TrackrAPI.Services.GestionExamen;
public class RespuestaService
{
    private readonly IRespuestaRepository respuestaRepository;

    public RespuestaService(
        IRespuestaRepository respuestaRepository)
    {
        this.respuestaRepository = respuestaRepository;
    }

    public RespuestaDto? ConsultarParaFormulario(int idRespuesta)
    {
        RespuestaDto? respuesta = respuestaRepository.ConsultarParaFormulario(idRespuesta);

        if (respuesta == null)
        {
            return null;
        }
        return respuesta;
    }

    public IEnumerable<RespuestaDto> ConsultarTodosPorReactivo(int idReactivo)
    {
        return respuestaRepository.ConsultarTodosPorReactivo(idReactivo);
    }

    public int Agregar(Respuesta respuesta)
    {
        respuestaRepository.Agregar(respuesta);
        return respuesta.IdRespuesta;
    }

    public void Editar(Respuesta respuesta)
    {
        respuestaRepository.Editar(respuesta);
    }

    public void Eliminar(int idRespuesta)
    {
        Respuesta respuesta = respuestaRepository.Consultar(idRespuesta);
        respuestaRepository.Eliminar(respuesta);
    }

}
